zebra-chain/src/parameters/network.rs

```diff
@@ -51,52 +53,105 @@ mod tests;
 /// after the grace period.
 const ZIP_212_GRACE_PERIOD_DURATION: HeightDiff = 32_256;
 
-/// An enum describing the possible network choices.
+/// Network consensus parameters for test networks such as Regtest and the default Testnet.
 #[derive(Clone, Debug, Default, Eq, PartialEq, Hash, Serialize, Deserialize)]
 #[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary))]
+pub struct NetworkParameters {}
+
+impl NetworkParameters {
+    /// Returns true if the instance of [`NetworkParameters`] represents the default public Testnet.
+    pub fn is_default_testnet(&self) -> bool {
+        self == &Self::default()
+    }
+}
+
+#[derive(Copy, Clone, Default, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+#[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary))]
+/// An enum describing the kind of network, whether it's the production mainnet or a testnet.
+pub enum NetworkKind {
```

Comment:
Add a correctness comment about the order of NetworkKind variants being important to bincode (de)serialization of the db format.